### PR TITLE
docs: update internationalization topics

### PR DESCRIPTION
**Pull request by josmar-crwdstffng**
Add new $localize and Ivy content.  Clean up navigation.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by petebacondarwin — COMMENTED** [review]
First pass through. I love the newly broken up files. Much easier to navigate.

File: aio/content/guide/i18n-common-format-data-locale.md [hunk 1]
```diff
@@ -1,18 +1,49 @@
...
+
+```typescript
+{{ amount | currency : 'en-US' }}
+```
+
+<div class="alert is-helpful">
+
+**NOTE**: The language-locale you set for `LOCALE_ID` token is ignored for the `CurrencyPipe`.
+
+`locale` parameter > `LOCALE_ID` token
```
[the remark this answers is not in the document]
Comment (josmar-crwdstffng) [reply aio/content/guide/i18n-common-format-data-locale.md:40]: Removed.

```md
`locale` parameter > `LOCALE_ID` token
```

Per our discussion, the relationship between the build locale, `LOCALE_ID`, `locale` parameter needs to be made clear in context of the way that Ivy handles localization.

**Review by josmar-crwdstffng — COMMENTED** [review]
@petebacondarwin 
I have addressed or commented on all of your initial suggestions.
Please verify and provide any additional details.

**Review by petebacondarwin — COMMENTED** [review]
Second round. Looking really good!

As agreed offline, I will check out the technical details for LOCAL_ID and global locale data tomorrow and come back with a view on those two pages.

**Review by josmar-crwdstffng — COMMENTED** [review]
@petebacondarwin 
Reviewed and implemented changes from 2nd TR.
Please confirm if any further changes are required.

**Review by petebacondarwin — APPROVED** [review]
LGTM from a technical perspective - just missed the change from `tick` to `backtick`.

As discussed offline - we can move the code snippets to external examples in a follow up PR.

**Review by josmar-crwdstffng — COMMENTED** [review]
@petebacondarwin 
Reviewed and implemented your latest suggestions.

**Review by TMDavisGoogle — CHANGES_REQUESTED** [review]
Light edits for punctuation, spelling. Inquiries regarding terminology.

**Review by josmar-crwdstffng — COMMENTED** [review]
@TMDavisGoogle 
I reviewed and addressed all of your comments.
Please review and either comment or approve.

**Review by mgechev — APPROVED** [review]

**Review by TMDavisGoogle — CHANGES_REQUESTED** [review]
Light edits (punctuation, capitalization) as well resolution of previous edits.

**Review by josmar-crwdstffng — COMMENTED** [review]
@TMDavisGoogle 
I reviewed and applied your latest editorial comments.
Review and either comment or approve the content.

**Review by TMDavisGoogle — APPROVED** [review]

**Pull request closed without merging**